include/midi.h

```diff
@@ -41,6 +41,33 @@ void MIDI_ListAll(Program *output_handler);
 void MIDI_RawOutByte(uint8_t data);
 void MIDI_ResumeSequence();
 
+enum class MessageType : uint8_t {
+	Channel,
+	Sysex,
+};
+
+struct MidiWork {
+	std::vector<uint8_t> message   = {};
+	uint16_t num_pending_frames    = 0;
+	MessageType message_type       = {};
+
+	// Default value constructor
+	MidiWork()                      = default;
+	MidiWork(MidiWork&&)            = default;
+	MidiWork& operator=(MidiWork&&) = default;
+
+	// Construct from movable values
+	MidiWork(std::vector<uint8_t>&& m, const uint16_t n, const MessageType t)
```

Comment:
minor: I think I like the underscore prefix style for the constructor args better.

Reply:
that's clearer, fixed.